vaadin-dev-server/src/main/java/com/vaadin/base/devserver/OpenInCurrentIde.java

```diff
@@ -0,0 +1,240 @@
+            printProcessTree(msg -> getLogger().debug(msg));
+            return false;
+        }
+
+        Info processInfo = maybeIdeCommand.get();
+        String cmd = processInfo.command().get();
+
+        if (isVSCode(processInfo)) {
+            return Open.open("vscode://file" + absolutePath + ":" + lineNumber);
+        } else if (isIdea(processInfo)) {
+            if (OSUtils.isMac()) {
+                // On Mac, the idea:// protocol works...
+                return Open.open("idea://open?file=" + absolutePath + "&line="
+                        + lineNumber);
+            } else {
+                try {
+                    run(cmd, "--line", lineNumber + "", absolutePath);
+                } catch (Exception e) {
+                    getLogger().error("Unable to launch IntelliJ IDEA", e);
+                }
+            }
+        } else if (isEclipse(processInfo)) {
+            if (OSUtils.isMac()) {
+                cmd = getBinary(processInfo);
+                try {
+                    run("open", "-a", cmd, absolutePath);
+                } catch (Exception e) {
+                    getLogger().error("Unable to launch Eclipse", e);
+                }
+            } else {
+                try {
+                    run(cmd, absolutePath + ":" + lineNumber);
+                } catch (Exception e) {
+                    getLogger().error("Unable to launch Eclipse", e);
+                }
+            }
+            return true;
+        }
+
+        return false;
+
+    }
+
+    static String getBinary(Info info) {
+        String cmd = info.command().get();
+        if (isIdea(info)) {
+            return getIdeaBinary(info);
+        } else if (isEclipse(info)) {
+            return cmd.replaceFirst("/Contents/MacOS/eclipse$", "");
+        }
+        return cmd;
+    }
+
+    private static Logger getLogger() {
+        return LoggerFactory.getLogger(OpenInCurrentIde.class);
+    }
+
+    public static void main(String[] args) {
+        // This is so it will be easier to debug problems in the future
+        printProcessTree(System.out::println);
+    }
+
+    private static void printProcessTree(Consumer<String> printer) {
+        for (Info info : getProcessTree()) {
+            printer.accept("Process tree:");
+            info.command()
+                    .ifPresent(value -> printer.accept("Command: " + value));
+            info.commandLine().ifPresent(
+                    value -> printer.accept("Command line: " + value));
+            info.arguments().ifPresent(values -> {
+                for (int i = 0; i < values.length; i++) {
+                    printer.accept("Arguments[" + i + "]: " + values[i]);
+                }
+            });
+            printer.accept("");
+        }
+
+    }
+
+    private static void run(String command, String... arguments)
+            throws Exception {
+        List<String> cmd = new ArrayList<>();
+        cmd.add(command);
```

Comment:
We should split `command` into separated tokens to avoid potential exceptions (e.g. `java.io.IOException: error=36, File name too long`)

```suggestion
        cmd.addAll(Arrays.asList(command.split("\\s")));
```

Reply:
Command here should just be the binary, not all the arguments